src/main/java/org/akaza/openclinica/service/ImportServiceImpl.java

```diff
@@ -1432,90 +1305,115 @@ private Object validateItem(ImportItemDataBean itemDataBean, CrfBean crf, EventC
+                }
+                else
+                    queryBean.setComment("");
+                discNote = (DiscrepancyNote) queryService.createQuery(helperBean, queryBean, isParentDn);
+                if (isParentDn)
+                    discNote.setThreadUuid(UUID.randomUUID().toString());
+                else {
+                    discNote.setThreadUuid(parentDn.getThreadUuid());
+                    discNote.setParentDiscrepancyNote(parentDn);
+                    helperBean.setParentDn(parentDn);
+                }
+                helperBean.setDn(discNote);
+                discNote = discrepancyNoteDao.saveOrUpdate(discNote);
+                queryService.saveQueryItemDatamap(helperBean);
+                DataImportReport dataImportReport =  new DataImportReport(studySubject.getOcOid(), studySubject.getLabel(),
+                        eventCrf.getStudyEvent().getStudyEventDefinition().getOc_oid(), eventCrf.getStudyEvent().getSampleOrdinal().toString(),
+                        eventCrf.getFormLayout().getCrf().getOcOid(), itemGroupDataBean.getItemGroupOID(), itemGroupDataBean.getItemGroupRepeatKey(), itemOid, discNote.getDisplayId()+" "+INSERTED, sdf_logFile.format(new Date()),  null);
+                dataImportReports.add(dataImportReport);
+            }
+            if (isParentDn)
+            {
+                for(int i = 0 ; i < discrepancyNoteBean.getChildNotes().size();i++){
+                    DiscrepancyNote childNote = createQuery(discrepancyNoteBean, study, studySubject, eventCrf, itemOid, itemGroupDataBean, itemData, discrepancyNoteBean.getChildNotes().get(i), discNote, false, dataImportReports);
+                    if(i == discrepancyNoteBean.getChildNotes().size()-1){
+                        discNote.setResolutionStatus(childNote.getResolutionStatus());
+                        discNote.setUserAccount(childNote.getUserAccount());
+                        discNote.setDetailedNotes(childNote.getDetailedNotes());
+                        discNote.setUserAccountByOwnerId(childNote.getUserAccountByOwnerId());
+                        discNote = discrepancyNoteDao.saveOrUpdate(discNote);
+                        if(!isNoteCreated){
+                            DataImportReport dataImportReport =  new DataImportReport(studySubject.getOcOid(), studySubject.getLabel(),
+                                    eventCrf.getStudyEvent().getStudyEventDefinition().getOc_oid(), eventCrf.getStudyEvent().getSampleOrdinal().toString(),
+                                    eventCrf.getFormLayout().getCrf().getOcOid(), itemGroupDataBean.getItemGroupOID(), itemGroupDataBean.getItemGroupRepeatKey(), itemOid, discNote.getDisplayId()+" "+UPDATED, sdf_logFile.format(new Date()),  null);
+                            dataImportReports.add(dataImportReport);
+                        }
+                    }
+                }
+            }
+        } catch (Exception e) {
+            logger.error("Query Import Failed: ", e);
```

Comment:
Translate the errors from queryService to a more user-friendly error messages

Reply:
I checked the query service code, The exception is not mandatory in those places, It was added to throw error if the container doesn't have the value required, I will remove the throws Exception, Since it will anyways get caught in the parent call.